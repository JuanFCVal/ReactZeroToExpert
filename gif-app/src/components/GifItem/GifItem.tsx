import React, { FC } from 'react'
import { IGifItem } from '../../interfaces/gifItem'
interface IGifItemProps{
    gif: IGifItem
}
const GifItem: FC<IGifItemProps> = ({gif}) => {
  return (
    <div className='card'>
    <img src={gif.url} alt={gif.title}/>
    <p>{gif.title}</p>
    </div>
  )
}

export default GifItem