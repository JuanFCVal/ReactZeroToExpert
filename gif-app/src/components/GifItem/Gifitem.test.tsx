import { render, screen } from '@testing-library/react';
import GifItem from './GifItem';
import '@testing-library/jest-dom'
describe('GifItem', () => {
    const gif = {
        id: '123',
        title: 'test',
        url: 'http://test.com'
    }
  it('Debe hacer match con el snapshot', () => {
    const { container } = render(<GifItem gif= {gif}/>);
    expect(container).toMatchSnapshot();
  });
  it('Debe renderizar correctamente', () => {
    const { getByText, getByRole } = render(<GifItem gif= {gif}/>);
    expect(getByRole('img')).toHaveAttribute('src', gif.url);
    expect(getByText(gif.title)).toBeInTheDocument();
  });
});