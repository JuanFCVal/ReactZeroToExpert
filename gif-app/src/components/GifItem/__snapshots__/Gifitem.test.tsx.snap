// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="test"
      src="http://test.com"
    />
    <p>
      test
    </p>
  </div>
</div>
`;
